mlflow/utils/environment.py

```diff
@@ -18,6 +25,124 @@
+            )
+        if dependencies is not None and not isinstance(dependencies, list):
+            raise TypeError(f"`dependencies` must be a list but got {type(dependencies)}")
+
+        self.python = python
+        self.build_dependencies = build_dependencies
+        self.dependencies = dependencies
+
+    def __str__(self):
+        return str(self.to_dict())
+
+    @classmethod
+    def default(cls):
+        return cls(
+            python=cls.get_current_python(),
+            build_dependencies=cls.get_default_build_dependencies(),
+            dependencies=[f"-r {_REQUIREMENTS_FILE_NAME}"],
+        )
+
+    @staticmethod
+    def get_current_python():
+        return ".".join(map(str, sys.version_info[:3]))
+
+    @staticmethod
+    def _get_package_version(package_name):
+        try:
+            return __import__(package_name).__version__
+        except (ImportError, AttributeError):
+            return None
+
+    @staticmethod
+    def get_default_build_dependencies():
+        build_dependencies = []
+        for package in ["pip", "setuptools", "wheel"]:
+            version = PythonEnv._get_package_version(package)
+            dep = (package + "==" + version) if version else package
+            build_dependencies.append(dep)
+        return build_dependencies
+
+    def to_dict(self):
+        return self.__dict__.copy()
+
+    @classmethod
+    def from_dict(cls, dct):
+        return cls(**dct)
+
+    def to_yaml(self, path):
+        with open(path, "w") as f:
+            # Exclude None and empty lists
+            data = {k: v for k, v in self.to_dict().items() if v}
+            yaml.safe_dump(data, f, sort_keys=False, default_flow_style=False)
+
+    @classmethod
+    def from_yaml(cls, path):
+        with open(path) as f:
+            return cls.from_dict(yaml.safe_load(f))
+
+    @classmethod
+    def from_conda_yaml(cls, path):
+        with open(path) as f:
+            conda_env = yaml.safe_load(f)
+
+        python = None
+        dependencies = None
+        build_dependencies = []
+        for dep in conda_env.get("dependencies", []):
+            if isinstance(dep, str):
+                if not python:
+                    match = _PYTHON_REGEX.match(dep)
+                    if match:
+                        python = match.group(2)
+                        continue
+
+                match = _BUILD_PACKAGE_REGEX.match(dep)
+                if match:
+                    name, operator, version = match.groups()
+                    operator = "==" if operator == "=" else operator
+                    build_dependencies.append(name + operator + version)
+
+            elif _is_pip_deps(dep):
+                dependencies = dep["pip"]
+
+        if python and dependencies:
+            return cls(
+                python=python,
+                build_dependencies=build_dependencies or None,
+                dependencies=dependencies,
+            )
+
+        raise ValueError(
```

Comment:
`MlflowException`. If this is due to invalid parameters, it should have code `INVALID_PARAMETER_VALUE`. Is it possible to use a more specific error message?